examples/python/gui_demo.py

```diff
@@ -219,114 +248,93 @@ def create_tree_widget(self, parent_frame):
 
         popup = tk.Menu(tree, tearoff=0)
         popup.add_command(label="Remove")
-        popup.add_command(label="Rename")
+        # popup.add_command(label="Rename")
+        popup.add_command(label="Connect")
         self.tree_popup = popup
 
-        self.functions_node_id = 'root_function_blocks'
+    def tree_update(self, new_selected_node=None):
+        # make sure the tree is empty
+        self.tree.delete(*self.tree.get_children())
 
+        self.selected_node = new_selected_node
 
-    def create_property_widget(self, parent_frame):
-        frame = tk.Frame()
-        # define columns
-        tree = ttk.Treeview(frame, columns=('value', 'unit', 'access'), show='tree headings')
+        self.tree_traverse_components_recursive(self.instance)
+        self.properties_update()
+        self.attributes_update()
+        self.selected_node = self.tree_restore_selection(
+            self.selected_node)  # reset in case the selected node outdates
 
-        scroll_bar = ttk.Scrollbar(frame, orient="vertical", command=tree.yview)
-        tree.configure(yscrollcommand=scroll_bar.set)
-        scroll_bar.pack(side="right", fill="y")
-        tree.pack(fill="both", expand=True)
+    def tree_traverse_components_recursive(self, component):
+        self.tree_add_component(
+            '' if component.parent is None else component.parent.global_id, component)
+        if (daq.IFolder.can_cast_from(component)):
```

Comment:
Check if component is folder and if it is empty (has no children) do not add it.